lib/index.js

```diff
@@ -363,6 +352,86 @@ module.exports = (bookshelf) => {
                 .where(wherePOUUID)
                 .del();
         },
+
+        bulkUpdate: function(rows, { transacting, returnInserted, columns } = {}) {
+            if (!Array.isArray(rows)) {
+                return Promise.reject(new Error('`rows` is not an array'));
+            }
+            const model = new this();
+            const idAttribute = model.idAttribute || 'id';
+
+            if (returnInserted && (_.isEmpty(model.orderedUuids) || !(idAttribute in model.orderedUuids))) {
+                return Promise.reject(new Error('Could not return bulk-inserted rows with autoincremented PK in MySQL'));
+            }
+
+            function validateRows(rows) {
+                if (!(model.validate && model.validateSave)) {
+                    return;
+                }
+                const validationErrors = rows
+                    .map((row) => {
+                        try {
+                            const rowToValidate = _.pickBy(row, (v,k) => {
+                                if (columns) {
+                                    if (!columns.includes(k)) {
+                                        return false;
+                                    }
+                                }
+                                return v !== null;
```

Comment:
Wouldn't this block setting null values?

Reply:
will check this out